### ISSUE TITLE
browserify support

### ISSUE THREAD

**Issue by rtsao** [uber/zero-config#8]
zero-config doesn't work in browserified files

I was hoping to use zero-config on a SPA that will live on web-toolshed. 

cc @orbiteleven @lxe @sh1mmer @Raynos

**Comment by mlmorg** [comment]
What about [brfs](https://github.com/substack/brfs) for handling configuration loading client-side?

**Comment by Raynos** [comment]
@mlmorg I think browserifying our secrets files and serving that to browsers is a BAD IDEA (TM).

**Comment by lxe** [comment]
@mlmorg @Raynos I've been playing around with brfs transform. One limitation/feature of brfs, is that it only overrides the `fs.readFileSyncs` in the current project, and not in the nested modules.

Also config-chain is very difficult to browserify (browserify actually ends up failing on index.js due to lack of semicolons).

Also, there's no clean way that I know of to conditionally detect whether brfs needs to be loaded at all. It's kind of wasteful to load brfs for projects that don't need client-side loading.

Nonetheless, there's a way to make zero-config browserifiable by directly `fs.readFile`ing things in `config/browser/env.json` and appending that to the config chain object.

Reference PR pending.

**Comment by Raynos** [comment]
It should be noted that browser code probably wants a seperate entry point.

Namely a `browser.js` entry point.

We will have to refactor this code to break out the shared / browser / node modules.

**Comment by lxe** [comment]
Conveniently, config-chain is unable to load anything at all after browserifying (if no changes are made to zero-config and config-chain).

Something like this works:

```javascript
var zeroConfig = require('zero-config');
var fs = require('fs');

// This always loads an empty config object
var config = zeroConfig(__dirname);

// Add stuff to it
config.set(JSON.parse(fs.readFileSync(__dirname + '/config/browser/common.json')));

console.log(config.get());
```

**Comment by Raynos** [comment]
Let's not "conveniently" rely config-chain not loading anything.

Let's duplicate some more of the implementation to make the browser implementation leaner and less "works on my machine (tm)".

**Comment by lxe** [comment]
> Let's not "conveniently" rely config-chain not loading anything.

I completely agree with this. I'm just exploring options and reporting what I find. Just notifying that "there is a way"

**Comment by timoxley** [comment]
Ping. Did you come up with anything for this?

**Comment by Raynos** [comment]
As far as I know it's not implemented.
